packages/server/src/db/index.ts

```diff
@@ -1,412 +1,212 @@
-  }
-
-  async getDidForUser(username: string, host: string): Promise<string | null> {
-    const row = await this.db
-      .select('did')
-      .from('user_dids')
-      .where({ username, host })
-    if (row.length < 1) return null
-    return row[0].did
-  }
-
-  async getUsername(did: string): Promise<string | null> {
-    const row = await this.db.select('*').from('user_dids').where({ did })
-    if (row.length < 1) return null
-    return `${row[0].username}@${row[0].host}`
-  }
-
-  async isDidRegistered(did: string): Promise<boolean> {
-    const un = await this.getUsername(did)
-    return un !== null
-  }
-
-  async isNameRegistered(username: string, host: string): Promise<boolean> {
-    const did = await this.getDidForUser(username, host)
-    return did !== null
-  }
-
-  // REPO ROOTS
-  // -----------
-
-  async createRepoRoot(did: string, cid: CID): Promise<void> {
-    await this.db.insert({ did, root: cid.toString() }).into('repo_roots')
-  }
-
-  async updateRepoRoot(did: string, cid: CID): Promise<void> {
-    await this.db('repo_roots').where({ did }).update({ root: cid.toString() })
-  }
-
   async getRepoRoot(did: string): Promise<CID | null> {
-    const row = await this.db
-      .select('root')
-      .from('repo_roots')
-      .where('did', did)
-    return row.length < 1 ? null : CID.parse(row[0].root)
-  }
-
-  // SERVER SUBSCRIPTIONS
-  // -----------
-
-  async createSubscription(host: string, user: string): Promise<void> {
-    await this.db.insert({ host, user }).into('subscriptions')
-  }
-
-  async getSubscriptionsForUser(user: string): Promise<string[]> {
-    const res = await this.db
-      .select('host')
-      .from('subscriptions')
-      .where({ user })
-    return res.map((row) => row.host)
-  }
-
-  // POSTS
-  // -----------
-
-  async getPost(
-    tid: string,
-    author: string,
-    namespace: string,
-  ): Promise<Post | null> {
-    const row = await this.db('posts')
-      .select('*')
-      .where({ tid, author, namespace })
-    if (row.length < 1) return null
-    return row[0]
-  }
-
-  async createPost(post: Post, cid: CID): Promise<void> {
-    await this.db('posts').insert({
-      ...post,
-      cid: cid.toString(),
-    })
-  }
-
-  async updatePost(post: Post, cid: CID): Promise<void> {
-    const { tid, author, namespace, text, time } = post
-    await this.db('posts')
-      .where({ tid, author, namespace })
-      .update({ text, time, cid: cid.toString() })
-  }
-
-  async deletePost(
-    tid: string,
-    author: string,
-    namespace: string,
-  ): Promise<void> {
-    await this.db('posts').where({ tid, author, namespace }).del()
-  }
-
-  async postCount(author: string): Promise<number> {
-    const res = await this.db('posts').count('*').where({ author })
-    const count = (res[0] || {})['count(*)']
-    if (typeof count !== 'number') {
-      throw new ServerError(500, 'Unable to retrieve post count')
+    const table = this.db.getRepository(RepoRoot)
```

Comment:
lol bad luck that TypeORM calls those "repositories"

Reply:
haha I know 😅